src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -408,14 +411,39 @@ public ResponseEntity<Participation> updateParticipation(@PathVariable long exer
         var originalParticipation = studentParticipationRepository.findByIdElseThrow(participation.getId());
         var user = userRepository.getUserWithGroupsAndAuthorities();
         authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.TEACHING_ASSISTANT, originalParticipation.getExercise(), null);
-        if (participation.getPresentationScore() == null || participation.getPresentationScore() < 0) {
-            participation.setPresentationScore(0);
+
+        Course course = findCourseFromParticipation(participation);
+        if (participation.getPresentationScore() != null && participation.getExercise().getPresentationScoreEnabled() != null
+                && participation.getExercise().getPresentationScoreEnabled()) {
+            Optional<GradingScale> gradingScale = gradingScaleService.findGradingScaleByCourseId(participation.getExercise().getCourseViaExerciseGroupOrCourseMember().getId());
+
+            // Validity of presentationScore for basic presentations
+            if (course.getPresentationScore() != null && course.getPresentationScore() > 0) {
+                if (participation.getPresentationScore() >= 1.) {
+                    participation.setPresentationScore(1.);
+                }
+                else {
+                    participation.setPresentationScore(null);
+                }
+            }
+            // Validity of presentationScore for graded presentations
+            if (gradingScale.isPresent() && gradingScale.get().getPresentationsNumber() != null) {
+                if (participation.getPresentationScore() > 100.) {
+                    participation.setPresentationScore(100.);
+                }
+                if (participation.getPresentationScore() < 0.) {
+                    participation.setPresentationScore(0.);
+                }
```

Comment:
I'd prefer an Exception here, otherwise the user might be confused why their chosen number was changed without an error